run test on 3.5

Carreau opened the pull request:
This should run test on python 3.5, but should not allow builds to be marked as failed.

Carreau committed:
run test on 3.5

takluyver commented:
`/home/travis/build.sh: line 41: /home/travis/virtualenv/pythonnightly/bin/activate: No such file or directory` - looks like Travis doesn't support this yet.

Carreau commented:
weird, it does work [for me on another project](https://travis-ci.org/Carreau/morning/jobs/60455753)

takluyver commented:
You're not using `sudo: false` on that one, perhaps that's the difference?

Carreau commented:
Sudo false is not needed anymore, any recently enough created repo run using docker unless asked otherwise. rerun, it "works" in the sens that we seem to have real faillures of tests.

Carreau commented:
And indeed. oinfo on `sum` in 3.5 returns more informations !

Carreau commented:
Should I move the DerivedInterrupt class out of locals at module level to fix test faillure ?

takluyver commented:
that seems like the easiest way to do it, yeah.

Carreau committed:
move custom exception at module.

fix test on python3.5+ where the qualified name include methods name
as well as <local> if defined in local scope.

Carreau commented:
> that seems like the easiest way to do it, yeah.

Done.

Carreau commented:
Travis is not unhappy anymore.

takluyver commented:
Well, it [still fails](https://travis-ci.org/ipython/ipython/jobs/60463265), but #8390 or something similar would fix that.

takluyver commented:
Tada! https://travis-ci.org/ipython/ipython/jobs/60782452
